sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java

```diff
@@ -385,9 +399,65 @@ private synchronized void setInstanceAutoPageTrackingEnabled(boolean autoPageTra
     /**
      * Implements {@link #setListener(AnalyticsListener)}.
      */
-    private synchronized void setInstanceListener(AnalyticsListener listener)
-    {
+    private synchronized void setInstanceListener(AnalyticsListener listener) {
         mAnalyticsListener = listener;
     }
 
+    /**
+     * Validates name.
+     *
+     * @param name log name to validate.
+     * @return <code>true<code/> if validation succeeds, otherwise <code>false</code>.
+     */
+    private static boolean validateName(String name) {
+        final int maxNameLength = 256;
+        if (name == null || name.isEmpty()) {
+            MobileCenterLog.error(Analytics.LOG_TAG, "Name cannot be null or empty.");
+            return false;
+        }
+        if (name.length() > maxNameLength) {
+            MobileCenterLog.error(Analytics.LOG_TAG, "Name length cannot be longer then " + maxNameLength + " characters.");
```

Comment:
Please print `Event '{name}' : name cannot be longer than...`  (typo with then by the way).